gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/data/repository/AvatarRepository.kt

```diff
@@ -84,14 +95,13 @@ internal class AvatarRepository(
         val token = tokenStorage.getToken(email.hash().toString())
         token?.let {
             when (val result = avatarService.updateAvatarCatching(avatarId, token, rating, altText)) {
-                is GravatarResult.Success -> GravatarResult.Success(result.value)
+                is GravatarResult.Success -> {
+                    avatarStorage.updateAvatar(email, result.value)
+                    GravatarResult.Success(result.value)
+                }
+
                 is GravatarResult.Failure -> GravatarResult.Failure(QuickEditorError.Request(result.error))
             }
         } ?: GravatarResult.Failure(QuickEditorError.TokenNotFound)
     }
 }
-
-internal data class EmailAvatars(
```

Comment:
`AvatarRepository` will work only with `Avatar`s, the viewmodel will map the Avatars to EmailAvatars, so moving this class to the PickersViewModel.